### PR TITLE
Fixed collection move with in the pathway should be captured like move e...

### PR DESCRIPTION
**Pull request by yamunaBQ**
...vent structure

### CODE REVIEW

File: api/src/main/java/org/ednovo/gooru/domain/service/classpage/ClasspageServiceImpl.java [hunk 1]
```diff
@@ -899,6 +899,8 @@ public Collection createPathway(String classId, Collection pathway, String colle
 	@Override
 	public Collection updatePathway(String classId, String pathwayGooruOid, Collection newPathway, User user) throws Exception {
 		Collection pathwayCollection = this.getCollectionRepository().getCollectionByIdWithType(pathwayGooruOid, ResourceType.Type.PATHWAY.getType());
+		final JSONObject itemData = new JSONObject();
```
Comment (sheebans) [inline api/src/main/java/org/ednovo/gooru/domain/service/classpage/ClasspageServiceImpl.java]: final JSONObject itemData = new JSONObject();  remove this unused  initialization..

**Comment by yamunaBQ** [comment]
removed unused initialization.

@sheebans

File: api/src/main/java/org/ednovo/gooru/domain/service/classpage/ClasspageServiceImpl.java [hunk 2]
```diff
@@ -1021,8 +1023,9 @@ public CollectionItem pathwayItemMoveWithReorder(String classId, String pathwayG
 		getAsyncExecutor().deleteFromCache("v2-class-data-"+ classId +"*");
 		try { 
 			
-		    this.getClasspageEventlog().getEventLogs(sourceItem.getResource().getGooruOid(), targetItem != null ? targetItem : collectionItem, pathwayGooruOid, user, sourceItem);
+		    this.getClasspageEventlog().getEventLogs(sourceItem.getResource().getGooruOid(), targetItem != null ? targetItem : collectionItem, pathwayGooruOid, user, sourceItem, targetItem);
 		} catch (JSONException e) {
```
Comment (sheebans) [inline api/src/main/java/org/ednovo/gooru/domain/service/classpage/ClasspageServiceImpl.java]: Move the try catch block in event log method,  and include the exception in logger @yamunaBQ

File: api/src/main/java/org/ednovo/gooru/domain/service/eventlogs/ClasspageEventLog.java [hunk 3]
```diff
@@ -158,15 +158,19 @@ public void getEventLogs(CollectionItem collectionItem, String pathwayId,User us
...
 	
-	public void getEventLogs(String collectionId, CollectionItem collectionItem, String pathwayId,User user, CollectionItem sourceItem) throws JSONException {
+	public void getEventLogs(String collectionId, CollectionItem collectionItem, String pathwayId,User user, CollectionItem sourceItem, CollectionItem targetItem) throws JSONException {
 	    SessionContextSupport.putLogParameter(EVENT_NAME, ITEM_EDIT);
 	    JSONObject context = SessionContextSupport.getLog().get(CONTEXT) != null ? new JSONObject(SessionContextSupport.getLog().get(CONTEXT).toString()) : new JSONObject();
 	    context.put(CONTENT_GOORU_ID, collectionId);
 	    context.put(PARENT_GOORU_ID, pathwayId);
-	    context.put(CONTENT_ITEM_ID, sourceItem.getCollectionItemId());
+	    context.put(TARGET_ITEM_ID, targetItem != null ? targetItem.getCollectionItemId() : null);
+	    context.put(CONTENT_ITEM_ID, sourceItem != null ? sourceItem.getCollectionItemId() : null);
```
Comment (sheebans) [inline api/src/main/java/org/ednovo/gooru/domain/service/eventlogs/ClasspageEventLog.java]: + context.put(CONTENT_ITEM_ID, sourceItem != null ? sourceItem.getCollectionItemId() : null);
+	    context.put(SOURCE_ITEM_ID, sourceItem != null ? sourceItem.getCollectionItemId() : null);

CONTENT_ITEM_ID, SOURCE_ITEM_ID Why both the ids are same ?

**Comment by yamunaBQ** [comment]
Fixed sourceItemId and contentItemId, changed exception in logger.

File: api/src/main/java/org/ednovo/gooru/domain/service/classpage/ClasspageServiceImpl.java [hunk 4]
```diff
@@ -899,6 +904,7 @@ public Collection createPathway(String classId, Collection pathway, String colle
 	@Override
 	public Collection updatePathway(String classId, String pathwayGooruOid, Collection newPathway, User user) throws Exception {
 		Collection pathwayCollection = this.getCollectionRepository().getCollectionByIdWithType(pathwayGooruOid, ResourceType.Type.PATHWAY.getType());
+		rejectIfNull(PATHWAY, GL0056, "pathway");
```
Comment (sheebans) [inline api/src/main/java/org/ednovo/gooru/domain/service/classpage/ClasspageServiceImpl.java]: "pathway" text as constant

Comment (yamunaBQ) [inline api/src/main/java/org/ednovo/gooru/domain/service/classpage/ClasspageServiceImpl.java]: changed pathway text as constant.

Comment (yamunaBQ) [inline api/src/main/java/org/ednovo/gooru/domain/service/classpage/ClasspageServiceImpl.java]: changed pathway text as constant.

@sheebans

**Comment by sheebans** [comment]
commit looks good to me

**Pull request merged**